targets/unity-v2/make.js

```diff
@@ -379,11 +379,11 @@ function getRequestActions(tabbing, apiCall, api) {
 
     if (api.name === "Client" && (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest"))
         return tabbing + "request.TitleId = request.TitleId ?? PlayFabSettings.TitleId;\n"
-            + tabbing + "if (request.TitleId == null) throw new Exception(\"Must be have PlayFabSettings.TitleId set to call this method\");\n";
+            + tabbing + "if (request.TitleId == null) throw new TitleIdNotSetException();\n";
```

Comment:
I like the idea of PlayFabException minimally wrapping Exception, but I'm not a fan of each of these individual specific exceptions. I'd rather change this:
throw new Exception(\"Must be have PlayFabSettings.TitleId set to call this method\");
to this:
throw new PlayFabException(\"Must be have PlayFabSettings.TitleId set to call this method\");

I think it accomplishes what you want, but it is still easy for me to add new exceptions where needed, without having to define new exception types all over.

Reply:
Sure, throwing PlayFabException is a lot better than throwing System.Exception.

The thing that I wath to achieve is also to be able to recover from some of exceptions that are not breaking internal state of PlayFabSdk.
For example,
```C#
try
{
  DoPlayfabRequest(arguments);
}
catch (TitleIdNotSetException)
{
  FixPlayfabTilleId();
}
catch (ClientNotLoggedInException)
{
  LoginPlayfabClient();
}
catch (Exception)
{
  SendReport();
  throw;
}
```

If you want, I can remove all strings from exception classes to make them as simple as possible.

If I still have not convince you to accept this change, I will leave only PlayFabException.

Reply:
The othe way suggested by @nitreo is to add exception code to PlayFabException. This will keep the only one class for exceptions and also provide a way to differentiate them in code.

Reply:
I like nitreo's suggestion, let's do that one.